torch/quantization/observer.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch.nn as nn
+import torch
+from functools import partial
+
+class Observer(nn.Module):
+    r"""Default Observer Module
+    A default implementation of the observer module, only works for
+    per_tensor_affine quantization scheme.
+    The module will record the running average of max and min value of the
+    observed Tensor and calulate_qparams will calculate the scale and zero_point
+
+    Other types of Observers should follow the same API, the first argument of
+    the constructor should be qoptions, and it can take arbitrary number of
+    arguments after that. and it should provide a calculate_qparam function.
+    TODO: Maybe add an abstract Observer class?
```

Comment:
TODO shouldn't be in the docstring but in the comment (we don't want to pollute docs)

Reply:
And having it as a abstract class is indeed a good idea

Reply:
Do we need an abstract class or a default class that everyone needs to inherit from (like nn.Module). Default observer would have a reference get stats/calcQparams, users can inherit and update.